test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1167,46 +1170,58 @@ public async Task GetBytes()
         }
 
         [Test]
-        public async Task GetStream([Values(true, false)] bool isAsync)
+        public async Task GetStreamSecondTimeThrows([Values(true, false)] bool isAsync)
         {
+            var expected = new byte[] { 1, 2, 3, 4, 5, 6, 7, 8 };
             var streamGetter = BuildStreamGetter(isAsync);
 
-            using (var conn = await OpenConnectionAsync())
+            using var conn = await OpenConnectionAsync();
+            using var cmd = new NpgsqlCommand($"SELECT {EncodeByteaHex(expected)}::bytea", conn);
+            using var reader = await cmd.ExecuteReaderAsync(Behavior);
+
+            await reader.ReadAsync();
+
+            using var stream = await streamGetter(reader, 0);
+
+            Assert.That(async () => await streamGetter(reader, 0),
+                Throws.Exception.TypeOf<InvalidOperationException>());
+        }
+
+        [Test]
+        public async Task GetStream([Values(true, false)] bool isAsync)
+        {
+            var binary = MemoryMarshal
+                .AsBytes<int>(Enumerable.Range(0, 1024).ToArray())
+                .ToArray();
+            await GetStream(binary, binary, "bytea");
+
+            var bigint = 0xDEADBEEFL;
+            var bigintBinary = BitConverter.GetBytes(
+                BitConverter.IsLittleEndian
+                ? BinaryPrimitives.ReverseEndianness(bigint)
+                : bigint);
+            await GetStream(bigint, bigintBinary);
+
+            async Task GetStream<T>(T value, byte[] expected, string? dataType = null)
```

Comment:
I think it's better to split this into two tests, one that tests byte[] and another which tests long. As you said before, it's better for a single test to test a single thing.